### PR TITLE
Adding gen_ancestry

### PR DESCRIPTION
**Pull request by slzarate**
#dxapp.json
- [x] Upgrade to 14.04
- [x] Add help and extension to input spec.
- [x] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [x] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [x] Use `regionalOptions` 
- [x] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [x] Add subjob restartability when applicable.
- [x] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Setting `"openSource": true`. 
- [x] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [x] Use AssetDepend instead of execDepend
- [x] Add default instance type that works for a general use case.
- [x] add new input if applicable

# Readme.md
- [x] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [ ] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [x] Basic testing using given input in Box or expected input
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.

### CODE REVIEW

**Comment by slzarate** [comment]
Sorry for the last-minute pull request. I thought I had created the pull request already.

File: sequencing/gen_ancestry/Readme.md [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
 # Determine Ancestry from PCA (DNAnexus Platform App)
 
- Determine Ancestry from PCA.  Uses an eigenvector file and training dataset listing known ancestries.  Runs QDA to determine posterior ancestries for all samples, even those in the training set.
+## What does this app do?
 
-This is the source code for an app that runs on the DNAnexus Platform.
-For more information about how to run or modify it, see
-https://wiki.dnanexus.com/.
-<!-- /dx-header -->
+This app determines from a PCA eigenvector file. It runs QDA (quadratic discriminant analysis) using a custom R script to determine posterior ancestries for all samples, even those in the training set.
```
Comment (jdidion) [inline sequencing/gen_ancestry/Readme.md]: "This app determines ancestry..."

File: sequencing/gen_ancestry/dxapp.json [hunk 2]
```diff
@@ -51,11 +53,49 @@
...
+      }
+    }
   },
-  "resources" : "project-BYpFk1Q0pB0xzQY8ZxgJFv1V",
   "openSource" : true,
-  "developers" : ["user-tnperson", "user-johnwallace123", "user-auv13", "user-atf3"],
-  "authorizedUsers": ["org-geisinger_affiliate", "org-geisinger", "org-support", "user-codushlaine", "user-semanti.mukherjee", "user-alexander.li", "user-shannon.bruse", "user-jan.freudenberg", "user-crisvh"]
+  "details": {
+    "upstreamVersion": "",
+    "whatsNew": ""
```
Comment (nainathangaraj) [inline sequencing/gen_ancestry/dxapp.json]: "whatsNew": "* 1.0.0 First version of the Determine Ancestry from PCA app."

File: sequencing/gen_ancestry/dxapp.json [hunk 3]
```diff
@@ -1,35 +1,38 @@
...
+      "help": "Self-reported ancestry file. Must have two columns: ID and RACE. IDs must match those in the eigenvector file.",
       "class": "file",
       "optional": false
     },
     {
       "name": "prefix",
-      "label": "PREFIX for ancestry file",
+      "label": "Prefix for ancestry file",
       "class": "string",
       "optional": true
```
Comment (jdidion) [inline sequencing/gen_ancestry/dxapp.json]: Add help

File: sequencing/gen_ancestry/dxapp.json [hunk 4]
```diff
@@ -41,7 +44,6 @@
       "optional": true,
       "default": 0.8
```
Comment (jdidion) [inline sequencing/gen_ancestry/dxapp.json:47]: Default needs to be an integer

File: sequencing/gen_ancestry/dxapp.json (hunk 2 above)
Comment (jdidion) [inline sequencing/gen_ancestry/dxapp.json]: Fill in whatsNew, something like "* 1.0.0: First gen_ancestry app release."

File: sequencing/gen_ancestry/dxapp.json [hunk 5]
```diff
@@ -1,35 +1,38 @@
...
-  "categories": [
+  "summary": "Determines ancestry from PCA eigenvector file using custom R script.",
+  "tags": [
     "Statistics"
   ],
+  "dxapi": "1.0.0",
+  "version": "1.0.0",
   "inputSpec": [
     {
       "name": "evec_fn",
```
Comment (yihchii) [inline sequencing/gen_ancestry/dxapp.json]: Maybe change the input parameter
"evec_fn" -> "pca_evec" to follow the input/output nomenclature?
Also will need to update the variable used on the app script.

File: sequencing/gen_ancestry/dxapp.json [hunk 6]
```diff
@@ -1,35 +1,38 @@
...
       "label": "PCA Eigenvector File",
+      "help": "Input PCA eigenvector file as generated by the vcf_to_plink app.",
       "class": "file",
-      "optional": false,
-      "patterns" : ["*.evec"]
+      "patterns": ["*.evec"],
+      "optional": false
     },
     {
       "name": "ancestry_fn",
```
Comment (yihchii) [inline sequencing/gen_ancestry/dxapp.json]: Maybe change the input parameter
"ancestry_fn" -> "ancestry_txt" to follow the input/output nomenclature?
Also will need to update the variable used on the app script.

File: sequencing/gen_ancestry/dxapp.json [hunk 7]
```diff
@@ -51,11 +53,49 @@
     }
```
Comment (yihchii) [inline sequencing/gen_ancestry/dxapp.json:55]: Maybe change the output parameter
"ancestry_out" -> "output_ancestry" to follow the input/output nomenclature?
Also will need to update the variable used on the app script.

File: sequencing/gen_ancestry/src/code.sh [hunk 8]
```diff
@@ -0,0 +1,36 @@
+set -x
+
```
Comment (yihchii) [inline sequencing/gen_ancestry/src/code.sh]: set -x -e -o pipefail

File: sequencing/gen_ancestry/dxapp.json [hunk 9]
```diff
@@ -1,35 +1,38 @@
 {
```
Comment (yihchii) [inline sequencing/gen_ancestry/dxapp.json:1]: Run dx-app-lint

**Review by yihchii — COMMENTED** [review]
LGTM! 
I added some comments regarding the nomenclature of the input/output variables. Also it would be nice to run `dx-app-lint` on the `dxapp.json` file.

File: sequencing/gen_ancestry/dxapp.json (hunk 4 above)
Comment (slzarate) [reply sequencing/gen_ancestry/dxapp.json:47]: No, it's type `float`

File: sequencing/gen_ancestry/dxapp.json (hunk 9 above)
Comment (slzarate) [reply sequencing/gen_ancestry/dxapp.json:1]: Done

**Pull request merged**